src/i2c.rs

```diff
@@ -98,10 +80,13 @@ impl<'a, Device: FtdiCommon> I2c<'a, Device> {
     /// ```no_run
     /// use ftdi_embedded_hal as hal;
     ///
-    /// let ftdi = hal::Ft232hHal::new()?.init_default()?;
-    /// let mut i2c = ftdi.i2c()?;
+    /// # #[cfg(feature = "libftd2xx")]
+    /// # {
+    /// let device = libftd2xx::Ft2232h::with_description("Dual RS232-HS A").unwrap();
+    /// let hal = hal::FtHal::init_freq(device, 3_000_000).unwrap();
+    /// let mut i2c = hal.i2c().unwrap();
     /// i2c.set_stop_start_len(10);
-    /// # Ok::<(), std::boxed::Box<dyn std::error::Error>>(())
+    /// # }
```

Comment:
Was there a reason to switch to `unwrap` without a return type?

Using `?` instead of `unwrap` is one of the API guidelines: https://rust-lang.github.io/api-guidelines/documentation.html#c-question-mark

Reply:
Ok. I dropped 'unwrap' in all the examples in favor of return type. Note that std::error:Error trait had to be implemented for the crate Error type in order to allow that.